### PR TITLE
PR from Dev

### PR DESCRIPTION
**Pull request by sebasbrowar**
# Description

This PR introduces a new feature: the `train_model` function located in `src/classifier.py`.  
It provides a reusable training pipeline for binary classification using logistic regression, including cross-validation (via GridSearchCV), validation evaluation, and logging to a file.

This enhancement is designed to make the model training process modular, testable, and production-ready, with clear logging and validation logic.

Dependencies:  
- Uses `pandas` for data preprocessing and manipulation (may be replaced with `polars` in future iterations for performance improvements).
- Requires `scikit-learn` for training and persistence.
- Logging is written to `classifier.log`.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

This PR is a foundational part of the Titanic classification system.  
Additional changes may follow, such as saving predictions to CSV, extending model options, or exporting model performance summaries.

# How has this been tested?

- A `pytest` fixture was added to train the model using preprocessed [Titanic data from Kaggle](https://www.kaggle.com/competitions/titanic/data).
- A test case (`test_classifier`) verifies the following:
  - A valid model object is returned.
  - The model implements the `predict` method.
  - The model achieves at least 80% accuracy on a hold-out validation split.

**Steps to reproduce**:
- Download the Titanic dataset from [Kaggle](https://www.kaggle.com/competitions/titanic/data).
- Extract `train.csv` and `test.csv` into a `data/` directory at the root of the project.
- Ensure that the file paths in `test_classifier.py` point to `data/train.csv` and `data/test.csv`.
- Run tests with `poetry run pytest -v`.
- Verify that `classifier.log` is created and contains model training and evaluation details.
- Confirm that the dataset is preprocessed correctly and that feature consistency is maintained between training and test sets.

# Checklist

- [X] Code has been formatted
- [X] Existing tests passed
- [X] New tests passed and were accepted
- [x] Logging is correctly set up
- [X] Linting checks are clean

### CODE REVIEW

File: src/classifier.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+logger.setLevel(logging.INFO)
+
+if not logger.handlers:
+    file_handler = logging.FileHandler("classifier.log")
+    formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+    file_handler.setFormatter(formatter)
+    logger.addHandler(file_handler)
+
+
+def train_model(X, y):
```
Comment (Pedro-Hdez) [inline src/classifier.py:18]: Integrate types for inputs and outputs (Reference: https://docs.python.org/3/library/typing.html)

Comment (Pedro-Hdez) [inline src/classifier.py]: Integrate types to inputs and outputs (Reference: https://docs.python.org/3/library/typing.html)

File: tests/test_classifier.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+from sklearn.model_selection import train_test_split
+
+from src.classifier import train_model
+
+# ==========================================
+# Paths to dataset (adjust if necessary)
+# ==========================================
+PROJECT_PATH = 'C:/Users/sbrxb/OneDrive/Documentos/GitHub/pq-starter-ds-project'
+TRAIN_PATH = PROJECT_PATH + '/data/train.csv'
+TEST_PATH = PROJECT_PATH + '/data/test.csv'
```
Comment (Pedro-Hdez) [inline tests/test_classifier.py]: Just to practice, could you handle these variables as environment variables from a .env file?

File: tests/test_classifier.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+# ==========================================
+# Load datasets
+# ==========================================
+ds_train = pd.read_csv(TRAIN_PATH)
+ds_test = pd.read_csv(TEST_PATH)
+
+# ==========================================
+# Preprocessing - Training Data
```
Comment (Pedro-Hdez) [inline tests/test_classifier.py]: Both training and test sets need the same preprocessing, to avoid duplicated code you can build a sklearn pipeline and apply it to both datasets (I think this is a good example if you haven't used it before: https://medium.com/@sahin.samia/scikit-learn-pipelines-explained-streamline-and-optimize-your-machine-learning-processes-f17b1beb86a4)

**Review by Pedro-Hdez — CHANGES_REQUESTED** [review]

File: src/classifier.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    Returns:
+    - GridSearchCV object ready to be fitted on training data.
+    """
+
+    # Complete Pipeline, preprocessing + model
+    logger.info("Creating Pipeline...")
+
+    pipeline = Pipeline([
+        ('preprocessor', preprocessor),
+        ('clf', LogisticRegression(class_weight='balanced', random_state=20))
```
Comment (Pedro-Hdez) [inline src/classifier.py:48]: It is a good idea to pass the classifier as a parameter

File: src/classifier.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+    # Log best parameters and CV score
+    logger.info(f"Best parameters found: {grid.best_params_}")
+    logger.info(f"Best cross-validation score: {grid.best_score_:.4f}")
+
+    # Get the best model from grid search
+    best_model = grid.best_estimator_
+
+    # Make predictions on the test set
+    y_test_preds = best_model.predict(X_test)
```
Comment (Pedro-Hdez) [inline src/classifier.py:106]: Also, it is a good idea to make train/test separate methods so we could test them individually and we will be adding more flexibility to the feature in case we need to validate multiple classifiers

**Review by Pedro-Hdez — CHANGES_REQUESTED** [review]
